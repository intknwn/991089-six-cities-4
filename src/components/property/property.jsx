import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import withReviewForm from '../../hocs/with-review-form/with-review-form.jsx';
import ReviewForm from '../review-form/review-form.jsx';
import Header from '../header/header.jsx';
import ReviewsList from '../reviews-list/reviews-list.jsx';
import Map from '../map/map.jsx';
import PlacesList from '../places-list/places-list.jsx';
import history from '../../history.js';
import {getRatingWidth} from '../../utils.js';
import {getLoadingStatus} from '../../reducer/app/selectors.js';
import {getPlaces, getNearby, getSortedByDateReviews} from '../../reducer/data/selectors.js';
import {getUser} from '../../reducer/user/selectors.js';
import {Operation} from '../../reducer/data/data.js';
import {
  placePropTypes,
  userPropTypes,
  reviewPropTypes,
  Place,
  PlaceCardType,
  AppRoute,
  bookmarkIconStyle
} from '../../const.js';

const ReviewFormWrapped = withReviewForm(ReviewForm);

class Property extends React.PureComponent {
  constructor(props) {
    super(props);
    this._place = null;
  }

  componentDidMount() {
    const {places, match, getPlaceReviews, getPlacesNearby} = this.props;

    const placeId = Number(match.params.id);
    const thisPlace = places.find((place) => place.id === placeId);
    if (thisPlace) {
      this._place = thisPlace;
      getPlaceReviews(placeId);
      getPlacesNearby(placeId);
    }
  }

  componentDidUpdate(prevProps) {
    const {
      places,
      getPlaceReviews,
      getPlacesNearby,
      match,
      location
    } = this.props;

    const placeId = Number(match.params.id);

    if (prevProps.location !== location || prevProps.places !== places) {
      const regex = /^\d+$/;

      if (!regex.test(match.params.id) || placeId > places.length) {
        history.push(AppRoute.ROOT);
        return;
      }

      this._place = places.find((place) => place.id === placeId);
      getPlaceReviews(placeId);
      getPlacesNearby(placeId);
    }

    if (prevProps.location !== location) {
      window.scrollTo(0, 0);
    }
  }

  _renderPlace() {
    const {
      user,
      reviews,
      placesNearby,
      onAddToFavoritesButtonClick,
      isLoading,
      postReview,
    } = this.props;

    const {
      bedrooms,
      description,
      goods,
      host,
      id,
      images,
      is_favorite: isFavorite,
      is_premium: isPremium,
      max_adults: maxAdults,
      price,
      rating,
      title,
      type,
    } = this._place;

    return (
      <div className="page">
        <Header user={user} />
        <main className="page__main page__main--property">
          <section className="property">
            <div className="property__gallery-container container">
              <div className="property__gallery">
                {images.slice(0, Place.MAX_GALLERY_IMAGES).map((src, i) => {
                  return (
                    <div key={i} className="property__image-wrapper">
                      <img
                        className="property__image"
                        src={src}
                        alt={title}
                      />
                    </div>
                  );
                })}
              </div>
            </div>
            <div className="property__container container">
              <div className="property__wrapper">
                {isPremium && <div className="property__mark"><span>Premium</span></div>}
                <div className="property__name-wrapper">
                  <h1 className="property__name">
                    {title}
                  </h1>
                  <button
                    className={`property__bookmark-button button ${isFavorite ? `property__bookmark-button--active` : ``}`}
                    onClick={() => onAddToFavoritesButtonClick(id, isFavorite)}
                    type="button"
                  >
                    <svg className="property__bookmark-icon" style={isFavorite ? bookmarkIconStyle : null} width={31} height={33}>
                      <use xlinkHref="#icon-bookmark" />
                    </svg>
                    <span className="visually-hidden">To bookmarks</span>
                  </button>
                </div>
                <div className="property__rating rating">
                  <div className="property__stars rating__stars">
                    <span style={getRatingWidth(rating)} />
                    <span className="visually-hidden">Rating</span>
                  </div>
                  <span className="property__rating-value rating__value">{rating}</span>
                </div>
                <ul className="property__features">
                  <li className="property__feature property__feature--entire">
                    {type}
                  </li>
                  <li className="property__feature property__feature--bedrooms">
                    {bedrooms}
                  </li>
                  <li className="property__feature property__feature--adults">
                    {maxAdults}
                  </li>
                </ul>
                <div className="property__price">
                  <b className="property__price-value">&euro;{price}</b>
                  <span className="property__price-text">&nbsp;night</span>
                </div>
                <div className="property__inside">
                  <h2 className="property__inside-title">What&apos;s inside</h2>
                  <ul className="property__inside-list">
                    {goods.map((good) => <li key={good} className="property__inside-item">{good}</li>)}
                  </ul>
                </div>
                <div className="property__host">
                  <h2 className="property__host-title">Meet the host</h2>
                  <div className="property__host-user user">
                    <div className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper">
                      <img
                        className="property__avatar user__avatar"
                        src={`/${host.avatar_url}`}
                        width={74}
                        height={74}
                        alt="Host avatar"
                      />
                    </div>
                    <span className="property__user-name">{host.name}</span>
                    {host.is_pro && <span className="property__user-status">Pro</span>}
                  </div>
                  <div className="property__description">
                    <p className="property__text">
                      {description}
                    </p>
                  </div>
                </div>
                <section className="property__reviews reviews">
                  <h2 className="reviews__title">
                    Reviews Â· <span className="reviews__amount">{reviews.length}</span>
                  </h2>
                  <ReviewsList reviews={reviews}/>
                  {user && <ReviewFormWrapped id={id} postReview={postReview} isLoading={isLoading}/>}
                </section>
              </div>
            </div>
            <section className="property__map map">
              <Map places={placesNearby} currentPlace={this._place}/>
            </section>
          </section>
          <div className="container">
            <section className="near-places places">
              <h2 className="near-places__title">
            Other places in the neighbourhood
              </h2>
              <div className="near-places__list places__list">
                <PlacesList
                  type={PlaceCardType.NEARBY}
                  places={placesNearby}
                />
              </div>
            </section>
          </div>
        </main>
      </div>
    );
  }

  render() {
    return this._place ? this._renderPlace() : <div></div>;
  }
}

Property.propTypes = {
  places: PropTypes.arrayOf(placePropTypes),
  placesNearby: PropTypes.arrayOf(placePropTypes),
  user: userPropTypes,
  reviews: PropTypes.arrayOf(reviewPropTypes),
  onAddToFavoritesButtonClick: PropTypes.func.isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string.isRequired
    })
  }),
  getPlaceReviews: PropTypes.func.isRequired,
  getPlacesNearby: PropTypes.func.isRequired,
  isLoading: PropTypes.bool.isRequired,
  postReview: PropTypes.func.isRequired,
  location: PropTypes.shape({
    hash: PropTypes.string,
    key: PropTypes.string,
    pathname: PropTypes.string,
    search: PropTypes.string,
    state: PropTypes.object,
  }),
};

const mapStateToProps = (state) => ({
  places: getPlaces(state),
  placesNearby: getNearby(state) || [],
  user: getUser(state),
  reviews: getSortedByDateReviews(state) || [],
  isLoading: getLoadingStatus(state),
});

const mapDispatchToProps = (dispatch) => ({
  onAddToFavoritesButtonClick(id, isFavorite) {
    dispatch(Operation.setFavorite(id, isFavorite));
  },
  getPlaceReviews(id) {
    dispatch(Operation.getReviews(id));
  },
  getPlacesNearby(id) {
    dispatch(Operation.getPlacesNearby(id));
  },
  postReview(review, onSuccess) {
    dispatch(Operation.postReview(review, onSuccess));
  },
});

export {Property};
export default connect(mapStateToProps, mapDispatchToProps)(Property);
