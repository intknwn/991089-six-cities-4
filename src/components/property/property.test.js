import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import {Router} from "react-router-dom";
import {Property} from './property.jsx';
import NameSpace from '../../reducer/name-space.js';
import history from '../../history.js';

const mockStore = configureStore([]);

const user = {
  "avatar_url": `img/1.png`,
  "email": `Oliver.conner@gmail.com`,
  "id": 1,
  "is_pro": false,
  "name": `Oliver.conner`,
};

const activeCity = {
  name: `Brussels`,
  location: {
    latitude: 50.846557,
    longitude: 4.351697,
    zoom: 13
  }
};

const places = [{
  "bedrooms": 3,
  "city": {
    "location": {
      "latitude": 50.846557,
      "longitude": 4.351697,
      "zoom": 13
    },
    "name": `Brussels`,
  },
  "description": `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country.`,
  "goods": [`Wahser`, `Fridge`, `Towels`, `Breakfast`, `Baby seat`, `Dishwasher`, `Laptop friendly workspace`, `Air conditioning`],
  "host": {
    "avatar_url": `img/avatar-angelina.jpg`,
    "id": 25,
    "is_pro": true,
    "name": `Angelina`
  },
  "id": 1,
  "images": [
    `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/20.jpg`,
    `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/7.jpg`,
  ],
  "is_favorite": false,
  "is_premium": false,
  "location": {
    "latitude": 50.842557,
    "longitude": 4.3536969999999995,
    "zoom": 16
  },
  "max_adults": 1,
  "preview_image": `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/20.jpg`,
  "price": 227,
  "rating": 2.2,
  "title": `Nice, cozy, warm big bed apartment`,
  "type": `room`
},
{
  "bedrooms": 2,
  "city": {
    "location": {
      "latitude": 50.846557,
      "longitude": 4.351697,
      "zoom": 13
    },
    "name": `Brussels`,
  },
  "description": `I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!`,
  "goods": [`Wahser`, `Fridge`, `Towels`, `Breakfast`, `Laptop friendly workspace`, `Air conditioning`],
  "host": {
    "avatar_url": `img/avatar-angelina.jpg`,
    "id": 25,
    "is_pro": true,
    "name": `Angelina`
  },
  "id": 2,
  "images": [
    `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/1.jpg`,
    `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/13.jpg`,
  ],
  "is_favorite": false,
  "is_premium": false,
  "location": {
    "latitude": 50.867557,
    "longitude": 4.357697,
    "zoom": 16
  },
  "max_adults": 3,
  "preview_image": `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/8.jpg`,
  "price": 327,
  "rating": 3.2,
  "title": `The house among olive`,
  "type": `house`
}];

const reviews = [{
  "id": 3,
  "user": {
    "id": 18,
    "is_pro": true,
    "name": `Sophie`,
    "avatar_url": `https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/9.jpg`
  },
  "rating": 3,
  "comment": `Beautiful space, fantastic location and atmosphere, really a wonderful place to spend a few days. Will be back.`,
  "date": `2020-07-17T16:06:01.820Z`
}, {
  "id": 2,
  "user": {
    "id": 19,
    "is_pro": false,
    "name": `Christina`,
    "avatar_url": `https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/10.jpg`
  },
  "rating": 4,
  "comment": `Home is amazing. It's like staying in a museum. The rooms, furnishings and artworks are incredible. The views of My Vesuvius`,
  "date": `2020-07-17T16:06:01.820Z`
}, {
  "id": 1,
  "user": {
    "id": 15,
    "is_pro": false,
    "name": `Kendall`,
    "avatar_url": `https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/6.jpg`
  },
  "rating": 4,
  "comment": `The house is very good, very happy, hygienic and simple living conditions around it are also very good. I hope to have the opportunity to come back. Thank you.`,
  "date": `2020-07-17T16:06:01.820Z`
}];

it(`Property render`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      places,
      placesNearby: places,
      reviews,
    },
    [NameSpace.APP]: {
      isLoading: false,
      activeCity
    },
    [NameSpace.USER]: {
      user
    },
  });

  const match = {
    params: {
      id: `1`,
    }
  };

  const tree = renderer
    .create(
        <Provider store={store}>
          <Router history={history}>
            <Property
              places={places}
              reviews={reviews}
              placesNearby={places}
              onAddToFavoritesButtonClick={() => {}}
              getPlaceReviews={() => {}}
              getPlacesNearby={() => {}}
              postReview={() => {}}
              isLoading={false}
              match={match}
            />
          </Router>
        </Provider>,
        {
          createNodeMock: () => document.createElement(`div`)
        })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
