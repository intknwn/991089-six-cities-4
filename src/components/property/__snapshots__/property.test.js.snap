// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Property render 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height={41}
              src="/img/logo.svg"
              width={81}
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/favorites"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                  style={Object {}}
                />
                <span
                  className="header__user-name user__name"
                >
                  Sign In
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Nice, cozy, warm big bed apartment"
              className="property__image"
              src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/20.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Nice, cozy, warm big bed apartment"
              className="property__image"
              src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/7.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              Nice, cozy, warm big bed apartment
            </h1>
            <button
              className="property__bookmark-button button "
              onClick={[Function]}
              type="button"
            >
              <svg
                className="property__bookmark-icon"
                height={33}
                style={null}
                width={31}
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "40%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            >
              2.2
            </span>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              room
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              3
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              1
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              227
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
              >
                Wahser
              </li>
              <li
                className="property__inside-item"
              >
                Fridge
              </li>
              <li
                className="property__inside-item"
              >
                Towels
              </li>
              <li
                className="property__inside-item"
              >
                Breakfast
              </li>
              <li
                className="property__inside-item"
              >
                Baby seat
              </li>
              <li
                className="property__inside-item"
              >
                Dishwasher
              </li>
              <li
                className="property__inside-item"
              >
                Laptop friendly workspace
              </li>
              <li
                className="property__inside-item"
              >
                Air conditioning
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
              >
                <img
                  alt="Host avatar"
                  className="property__avatar user__avatar"
                  height={74}
                  src="/img/avatar-angelina.jpg"
                  width={74}
                />
              </div>
              <span
                className="property__user-name"
              >
                Angelina
              </span>
              <span
                className="property__user-status"
              >
                Pro
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country.
              </p>
            </div>
          </div>
          <section
            className="property__reviews reviews"
          >
            <h2
              className="reviews__title"
            >
              Reviews · 
              <span
                className="reviews__amount"
              >
                3
              </span>
            </h2>
            <ul
              className="reviews__list"
            >
              <li
                className="reviews__item"
              >
                <div
                  className="reviews__user user"
                >
                  <div
                    className="reviews__avatar-wrapper user__avatar-wrapper"
                  >
                    <img
                      alt="Sophie"
                      className="reviews__avatar user__avatar"
                      height={54}
                      src="https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/9.jpg"
                      width={54}
                    />
                  </div>
                  <span
                    className="reviews__user-name"
                  >
                    Sophie
                  </span>
                </div>
                <div
                  className="reviews__info"
                >
                  <div
                    className="reviews__rating rating"
                  >
                    <div
                      className="reviews__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "60%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <p
                    className="reviews__text"
                  >
                    Beautiful space, fantastic location and atmosphere, really a wonderful place to spend a few days. Will be back.
                  </p>
                  <time
                    className="reviews__time"
                    dateTime="2020-07-17T16:06:01.820Z"
                  >
                    July 2020
                  </time>
                </div>
              </li>
              <li
                className="reviews__item"
              >
                <div
                  className="reviews__user user"
                >
                  <div
                    className="reviews__avatar-wrapper user__avatar-wrapper"
                  >
                    <img
                      alt="Christina"
                      className="reviews__avatar user__avatar"
                      height={54}
                      src="https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/10.jpg"
                      width={54}
                    />
                  </div>
                  <span
                    className="reviews__user-name"
                  >
                    Christina
                  </span>
                </div>
                <div
                  className="reviews__info"
                >
                  <div
                    className="reviews__rating rating"
                  >
                    <div
                      className="reviews__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <p
                    className="reviews__text"
                  >
                    Home is amazing. It's like staying in a museum. The rooms, furnishings and artworks are incredible. The views of My Vesuvius
                  </p>
                  <time
                    className="reviews__time"
                    dateTime="2020-07-17T16:06:01.820Z"
                  >
                    July 2020
                  </time>
                </div>
              </li>
              <li
                className="reviews__item"
              >
                <div
                  className="reviews__user user"
                >
                  <div
                    className="reviews__avatar-wrapper user__avatar-wrapper"
                  >
                    <img
                      alt="Kendall"
                      className="reviews__avatar user__avatar"
                      height={54}
                      src="https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/6.jpg"
                      width={54}
                    />
                  </div>
                  <span
                    className="reviews__user-name"
                  >
                    Kendall
                  </span>
                </div>
                <div
                  className="reviews__info"
                >
                  <div
                    className="reviews__rating rating"
                  >
                    <div
                      className="reviews__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <p
                    className="reviews__text"
                  >
                    The house is very good, very happy, hygienic and simple living conditions around it are also very good. I hope to have the opportunity to come back. Thank you.
                  </p>
                  <time
                    className="reviews__time"
                    dateTime="2020-07-17T16:06:01.820Z"
                  >
                    July 2020
                  </time>
                </div>
              </li>
            </ul>
          </section>
        </div>
      </div>
      <section
        className="property__map map"
      >
        <div
          style={
            Object {
              "height": "100%",
            }
          }
        />
      </section>
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <div
          className="near-places__list places__list"
        >
          <article
            className="near-places__card place-card"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="near-places__image-wrapper place-card__image-wrapper"
            >
              <a
                href="/place/1"
                onClick={[Function]}
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height={200}
                  src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/20.jpg"
                  width={260}
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    227
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button  button"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "40%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="/place/1"
                  onClick={[Function]}
                >
                  Nice, cozy, warm big bed apartment
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                room
              </p>
            </div>
          </article>
          <article
            className="near-places__card place-card"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="near-places__image-wrapper place-card__image-wrapper"
            >
              <a
                href="/place/2"
                onClick={[Function]}
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height={200}
                  src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/8.jpg"
                  width={260}
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    327
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button  button"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "60%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="/place/2"
                  onClick={[Function]}
                >
                  The house among olive
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                house
              </p>
            </div>
          </article>
        </div>
      </section>
    </div>
  </main>
</div>
`;
